Support whitespace between language codes

kissen opened the pull request:
So far, when configuring multiple spell check languages, Hexchat requires users to separate multiple entries with commas and only commas. This patch allows users to also enter whitespace, e.g.

    de_DE, en_US

as is common in many applications.

I believe this patch may resolve issue https://github.com/hexchat/hexchat/issues/2508. I know the current behavior of Hexchat was very confusing to me.

TingPing commented:
Can you test this:

`de_DE , en_US`

I suspect `en_US` will not be picked up. This bit of code uses `strtok()` which reads over the string and `g_strstrip()` modifies the string in-place.

I think the ideal fix for that is using `g_strsplit()` instead of `strtok()`.

kissen commented:
I tested `foo_FAKE, de_DE , en_US` and both `de_DE` and `en_US` are picked up correctly.

That is because `strtok` splits up `foo_FAKE, de_DE , en_US` into `foo_FAKE`, ` de_DE ` and ` en_US`, that is `strtok` inserts terminating zeroes in-place into the string it is passed. These tokens still contain whitespace which is then removed with `g_strstrip`.

`g_strstrip` just chains `g_strchug` and `g_strchomp`, both of which modify the string in-place, but always return the pointer they were passed to. So something like ` de_DE ` = `{' ', 'd', 'e', '_', 'D', 'E', ' ', 0 }` gets turned to ` de_DE ` = `{'d', 'e', '_', 'D', 'E', 0, 0, 0 }`.

That said, the [documentation for `g_strstrip(str)`](https://docs.gtk.org/glib/func.strstrip.html) does not grantee that it always returns `str`. I will re-write this patch to use `g_strsplit()` instead at next convenience.

TingPing commented:
> I will re-write this patch to use g_strsplit() instead at next convenience.

Thanks! The less `strtok()` is used the better :)

kissen commented:
I have updated the pull request to use `g_strsplit_set` rather than `strtok`. We now split on comma, spaces and tab characters. I think for most purposes that is enough.
